### PR TITLE
HIVE-25272: READ transactions are getting logged in NOTIFICATION LOG

### PR DESCRIPTION
**Pull request by pkumarsinha**

### CODE REVIEW

**Review by aasha — APPROVED** [review]

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
 /**
  * TestReplicationScenariosAcidTables - test replication for ACID tables.
  */
-@org.junit.Ignore("HIVE-25267")
+//@org.junit.Ignore("HIVE-25267")
```
Comment (pvary) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java:77]: Does this fix the flakiness issue mentioned in HIVE-25267?

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java [hunk 2]
```diff
@@ -176,6 +177,28 @@ public void testReplOperationsNotCapturedInNotificationLog() throws Throwable {
...
+    primary.dump(primaryDbName);
+    replica.run("REPL LOAD " + primaryDbName + " INTO " + replicatedDbName);
+    //Perform empty incremental dump and load so that all db level properties are altered.
+    primary.dump(primaryDbName);
+    replica.run("REPL LOAD " + primaryDbName + " INTO " + replicatedDbName);
+    primary.run("insert into " + primaryDbName + ".t1 values(1)");
+    long lastEventId = primary.getCurrentNotificationEventId().getEventId();
+    primary.run("DESCRIBE DATABASE " + primaryDbName );
+    primary.run("SELECT * from " + primaryDbName + ".t1");
+    primary.run("SHOW tables " + primaryDbName);
```
Comment (pvary) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java:194]: What is the reason behind running these commands but discarding the results?

Comment (pvary) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java:194]: never mind, I got it 😄

**Comment by pvary** [comment]
@deniskuzZ: This PR extends the scope of the READ-ONLY transactions. If you have time, could you please take a look?

Thanks,
Peter

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java (hunk 1 above)
Comment (pkumarsinha) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java:77]: Thanks for the  review @pvary . No, it doesn't. I wanted to get a feedback on this. Also, the current test were in this file only so had enabled it. I will revert this if I couldn't fix the flakiness as a part of JIRA

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java [hunk 3]
```diff
@@ -176,6 +177,28 @@ public void testReplOperationsNotCapturedInNotificationLog() throws Throwable {
     assert lastEventId == currentEventId;
   }
 
+  @Test
+  public void testREADOperationsNotCapturedInNotificationLog() throws Throwable {
```
[the remark this answers is not in the document]
Comment (pkumarsinha) [reply itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java]: Done

File: ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java [hunk 4]
```diff
@@ -3048,17 +3059,12 @@ public static void validateAcidPartitionLocation(String location, Configuration
    * @param tree AST
    */
   public static TxnType getTxnType(Configuration conf, ASTNode tree) {
-    final ASTSearcher astSearcher = new ASTSearcher();
-
+    int tp = tree.getToken().getType();
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java:3062]: doesn't seem to be used anywhere

File: ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java [hunk 5]
```diff
@@ -182,10 +183,20 @@ public boolean accept(Path path) {
   public static final int MAX_STATEMENTS_PER_TXN = 10000;
   public static final Pattern LEGACY_BUCKET_DIGIT_PATTERN = Pattern.compile("^[0-9]{6}");
   public static final Pattern BUCKET_PATTERN = Pattern.compile("bucket_([0-9]+)(_[0-9]+)?$");
+  private static final Set<Integer> READ_TXN_TOKENS = new HashSet<Integer>();
 
   private static Cache<String, DirInfoValue> dirCache;
   private static AtomicBoolean dirCacheInited = new AtomicBoolean();
 
+  static {
+    READ_TXN_TOKENS.addAll(Arrays.asList(
```
Comment (deniskuzZ) [inline ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java:192]: does it cover SHOW LOCKS and EXPLAIN LOCKS?

File: ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java (hunk 4 above)
Comment (pkumarsinha) [reply ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java:3062]: Oops!! Was using it while debugging something. Missed to remove it. Will do.

File: ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java (hunk 5 above)
Comment (pkumarsinha) [reply ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java:192]: Yes, they are covered

File: itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java [hunk 6]
```diff
@@ -176,6 +177,41 @@ public void testReplOperationsNotCapturedInNotificationLog() throws Throwable {
...
+      primary.run("USE " + dbName);
+      primary.run("DESCRIBE DATABASE " + dbName);
+      primary.run("DESCRIBE "+ dbName + ".t1");
+      primary.run("SELECT * FROM " + dbName + ".t1");
+      primary.run("SHOW TABLES " + dbName);
+      primary.run("SHOW TABLE EXTENDED LIKE 't1'");
+      primary.run("SHOW TBLPROPERTIES t1");
+      primary.run("EXPLAIN SELECT * from " + dbName + ".t1");
+      primary.run("SHOW LOCKS");
+      primary.run("EXPLAIN SHOW LOCKS");
```
Comment (deniskuzZ) [inline itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestReplicationScenariosAcidTables.java:206]: Could you please add test case for 'EXPLAIN LOCKS <query>' that is widely used?
```
EXPLAIN LOCKS UPDATE target SET b = 1 WHERE p IN (SELECT t.q1 FROM source t WHERE t.a1=5)'
```

**Review by aasha — APPROVED** [review]

**Pull request merged**